test/e2e/persistent_volumes.go

```diff
@@ -134,53 +134,95 @@ func createPVC(c *client.Client, ns string, pvc *api.PersistentVolumeClaim) (*ap
 	return pvc, nil
 }
 
-// Create a PV and PVC based on the passed in nfs-server ip and namespace.
-// There are 4 combinations, 3 of which are supported here:
-//   1) prebind and create pvc first
-//   2) no prebind and create pvc first
-//   3) no prebind and create pv first
-// The case of prebinding and creating the pv first is not possible due to using a
-// *generated* name in the pvc, and thus not knowing the claim's name until after
-// it has been created.
-// **Note: this function complements makePersistentVolume() and fills in the remaining
-// name field in the pv's ClaimRef.
-func createPVandPVC(c *client.Client, serverIP, ns string, pvFirst, preBind bool) (*api.PersistentVolume, *api.PersistentVolumeClaim, error) {
-
-	var bindTo *api.PersistentVolumeClaim
-	var err error
+// Create a PV followed by the PVC based on the passed in nfs-server ip and
+// namespace. There is no prebinding.
+func createPVPVCnoPrebind(c *client.Client, serverIP, ns string) (*api.PersistentVolume, *api.PersistentVolumeClaim, error) {
+
+	By("Creating the non-pre-bound PV followed by the PVC")
 
-	pvc := makePersistentVolumeClaim(ns) // pvc.Name not known yet
+	pv := makePersistentVolume(serverIP, nil)
+	pvc := makePersistentVolumeClaim(ns)
 
-	bindTo = nil
-	if preBind { // implies pvc *must* be created before the pv
-		pvFirst = false
-		bindTo = pvc
+	pv, err := createPV(c, pv)
+	if err != nil {
+		return nil, nil, err
 	}
-	pv := makePersistentVolume(serverIP, bindTo)
 
-	if pvFirst {
-		By("Creating the PV followed by the PVC")
-		pv, err = createPV(c, pv)
-	} else {
-		By("Creating the PVC followed by the PV")
-		pvc, err = createPVC(c, ns, pvc)
+	pvc, err = createPVC(c, ns, pvc)
+	if err != nil {
+		return pv, nil, err
 	}
+
+	return pv, pvc, nil
+}
+
+// Create a PVC followed by the PV based on the passed in nfs-server ip and
+// namespace. There is no prebinding.
+func createPVCPVnoPrebind(c *client.Client, serverIP, ns string) (*api.PersistentVolume, *api.PersistentVolumeClaim, error) {
```

Comment:
`createPVCPVnoPrebind` and `createPVCPVpreBind` are almost the same except few lines. Please make them one with a `prebind bool` parameter.